integer/src/rns.rs

```diff
@@ -600,6 +600,7 @@ impl<W: FieldExt, N: FieldExt, const NUMBER_OF_LIMBS: usize, const BIT_LEN_LIMB:
     }
 
     /// Computes the overflow that each component of the [`Rns`] must support.
+    // TODO: consider soundness of only single overflow length
```

Comment:
If we are worrying about the cost of extra `overflow_length`, I think we can use tagged table to make it constant cost, I have implemented a simple version [here](https://github.com/han0110/halo2wrong/blob/feature%2Frange-chip-with-tagged-table/maingate/src/range.rs) to reduce the gas cost for evm verifier.

Reply:
Hmm qq: can this overflow table also tagged into the limb table?

Reply:
yeah I think so, if we assume overflows are always smaller than limb. If not, we can just have 2 table, still a constant cost.